### PR TITLE
ConnectionManager: requesting a list of connections

### PR DESCRIPTION
**Pull request by jacek-izykowski**

### CODE REVIEW

File: python/veles/proto/connection.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from veles.schema import model, fields
+
+
+class Connection(model.Model):
+    client_id = fields.SmallUnsignedInteger()
+    client_name = fields.String()
+    client_type = fields.String()
```
Comment (koriakin) [inline python/veles/proto/connection.py]: MsgConnect has version and description, let's add it here as well

Comment (jacek-izykowski) [inline python/veles/proto/connection.py]: Fixed.

File: src/ui/connectionmanager.cc [hunk 2]
```diff
@@ -211,13 +213,45 @@ void ConnectionManager::updateConnectionStatus(
...
+        = new messages::MsgListConnections(
+        network_client_->nextQid(), true);
+    network_client_->sendMessage(client::msg_ptr(list_connections_message));
+  }
+}
+
+void ConnectionManager::messageReceived(client::msg_ptr message) {
+  if (message->object_type == "connections_reply") {
+    messages::MsgConnectionsReply* connections_reply
+        = dynamic_cast<messages::MsgConnectionsReply*>(message.get());
```
Comment (altran01) [inline src/ui/connectionmanager.cc]: `std::dynamic_pointer_cast` please.

File: src/ui/connectionmanager.cc [hunk 3]
```diff
@@ -211,13 +213,45 @@ void ConnectionManager::updateConnectionStatus(
...
+    if (connections_reply) {
+      QTextStream& out = *network_client_output_;
+      out << "ConnectionManager: received updated list of connections:"
+          << endl;
+
+      for (auto connection : *(connections_reply->connections.get())) {
```
Comment (altran01) [inline src/ui/connectionmanager.cc]: This `.get()` here is redundant - operator* is defined for shared_ptr.

File: python/veles/proto/connection.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+from veles.schema import model, fields
+
+
+class Connection(model.Model):
+    client_id = fields.SmallUnsignedInteger()
+    client_name = fields.String()
+    client_version = fields.String()
+    client_description = fields.String()
+    client_type = fields.String()
+
```
Comment (altran01) [inline python/veles/proto/connection.py]: pep8 check failed for this file.

File: python/veles/server/proto.py [hunk 5]
```diff
@@ -606,6 +607,12 @@ def send_msg(self, msg):
         del self.phids[msg.phid]
         self.send_msg(messages.MsgPluginHandlerUnregistered(phid=msg.phid))
 
+    async def msg_list_connections(self, msg):
+        self.send_msg(messages.MsgConnectionsReply(
+            qid=msg.qid,
+            connections=[]
```
Comment (altran01) [inline python/veles/server/proto.py]: Why don't you return list of connections here? Do you intend to do it in separate PR together with Python clients handling for those messages?

File: python/veles/server/proto.py [hunk 6]
```diff
@@ -296,6 +296,7 @@ def data_received(self, data):
                 'plugin_broadcast_result': self.msg_plugin_broadcast_result,
                 'plugin_handler_unregister':
                     self.msg_plugin_handler_unregister,
+                'list_connections': self.msg_list_connections
```
Comment (adamiwaniuk) [inline python/veles/server/proto.py]: please place comma after last element

File: python/veles/server/proto.py (hunk 5 above)
Comment (adamiwaniuk) [inline python/veles/server/proto.py]: please place comma after last element

File: src/ui/connectionmanager.cc [hunk 7]
```diff
@@ -211,13 +213,50 @@ void ConnectionManager::updateConnectionStatus(
...
+    if (connections_reply) {
+      QTextStream& out = *network_client_output_;
+      out << "ConnectionManager: received updated list of connections:"
+          << endl;
+
+      if (connections_reply->connections) {
+        for (auto connection : *connections_reply->connections) {
+          out
+              << "    id = " << connection->client_id
+              << " name = \"" << QString(connection->client_name->c_str())
```
Comment (adamiwaniuk) [inline src/ui/connectionmanager.cc]: maybe QString::fromStdString will be better ?

File: python/veles/server/proto.py (hunk 6 above)
Comment (jacek-izykowski) [inline python/veles/server/proto.py]: Done.

File: python/veles/server/proto.py (hunk 5 above)
Comment (jacek-izykowski) [inline python/veles/server/proto.py]: It's going to be a separate PR.

Comment (jacek-izykowski) [inline python/veles/server/proto.py]: Done.

File: src/ui/connectionmanager.cc (hunk 7 above)
Comment (jacek-izykowski) [inline src/ui/connectionmanager.cc]: Good point. Changed.
